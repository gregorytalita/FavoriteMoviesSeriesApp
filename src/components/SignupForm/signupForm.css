.formContent {
  display: flex;
  flex-direction: column;
  height: 175px;
  margin: 40px 0;
  justify-content: space-around;
}

.formButton {
  width: 100%;
  margin: 32px 24px;
}

