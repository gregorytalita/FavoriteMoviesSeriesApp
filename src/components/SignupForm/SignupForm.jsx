import React, { useState } from 'react'
import {
  Typography,
  InputLabel,
  FormControl,
  Input,
  Button
} from '@material-ui/core'

import './signupForm.css'

const SignupForm = ({ register, error }) => {
  const [email, setEmail] = useState('')
  const [password, setPassword] = useState()
  const [confirmPassword, setConfirmPassword] = useState()
  const [match, handleMatch] = useState(true)

  const submit = () => {
    const validPass = password !== ''
    const validConfirm = confirmPassword !== ''

    if (validPass && validConfirm) {
      register({ email, password })
      handleMatch(false)
    } else {
      handleMatch(true)
    }
  }

  return (
    <>
      <Typography variant='h1' align='center' style={{ fontSize: 32 }}>
        Signup
      </Typography>
      <div className='formContent'>
        {
          error &&
          <Typography>
            Seems that you typed something wrong :c
          </Typography>
        }
        <FormControl error={error}>
          <InputLabel htmlFor='email'>Email</InputLabel>
          <Input
            id='email'
            value={email}
            onChange={({ target }) => setEmail(target.value)}
            aria-describedby='email-text'
          />
        </FormControl>
        <FormControl error={error || !match}>
          <InputLabel htmlFor='password'>Password</InputLabel>
          <Input
            id='password'
            type='password'
            value={password}
            onChange={({ target }) => setPassword(target.value)}
            aria-describedby='password-text'
          />
        </FormControl>
        <FormControl error={error || !match}>
          <InputLabel htmlFor='confirm-password'>Confirm password</InputLabel>
          <Input
            id='confirm-password'
            type='password'
            value={confirmPassword}
            onChange={({ target }) => setConfirmPassword(target.value)}
            aria-describedby='confirm-password-text'
          />
        </FormControl>
      </div>
      <Button
        variant='outlined'
        color='secondary'
        onClick={submit}
        className='formButton'
      >
        Signup
      </Button>
    </>
  )
}

export default SignupForm
