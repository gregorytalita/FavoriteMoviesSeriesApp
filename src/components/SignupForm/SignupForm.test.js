import React from 'react'
import { create } from 'react-test-renderer'
import SignupForm from './SignupForm'

describe('SignupForm component', () => {
  test('it matches the snapshot', () => {
    const component = create(<SignupForm />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
