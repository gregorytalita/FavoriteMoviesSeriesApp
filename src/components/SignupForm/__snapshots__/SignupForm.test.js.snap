// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm component it matches the snapshot 1`] = `
Array [
  <h1
    className="MuiTypography-root MuiTypography-h1 MuiTypography-alignCenter"
    style={
      Object {
        "fontSize": 32,
      }
    }
  >
    Signup
  </h1>,
  <div
    className="formContent"
  >
    <div
      className="MuiFormControl-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
        htmlFor="email"
      >
        Email
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-describedby="email-text"
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
        htmlFor="password"
      >
        Password
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-describedby="password-text"
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="password"
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
        htmlFor="confirm-password"
      >
        Confirm password
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-describedby="confirm-password-text"
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="confirm-password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="password"
        />
      </div>
    </div>
  </div>,
  <button
    className="MuiButtonBase-root MuiButton-root formButton MuiButton-outlined MuiButton-outlinedSecondary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Signup
    </span>
  </button>,
]
`;
